src/VisualStudio/Core/Def/FindReferences/Entries/DefinitionItemEntry.cs

```diff
@@ -18,27 +24,45 @@ internal partial class StreamingFindUsagesPresenter
         /// GoToDefinition/FindImplementations.  In these operations, we don't want to 
         /// create a DefinitionBucket.  So we instead just so the symbol as a normal row.
         /// </summary>
-        private class DefinitionItemEntry : AbstractDocumentSpanEntry
+        private class DefinitionItemEntry(
+            AbstractTableDataSourceFindUsagesContext context,
+            RoslynDefinitionBucket definitionBucket,
+            string projectName,
+            Guid projectGuid,
+            SourceText lineText,
+            MappedSpanResult mappedSpanResult,
+            Document document,
+            IThreadingContext threadingContext)
+            : AbstractDocumentSpanEntry(context, definitionBucket, projectGuid, lineText, mappedSpanResult), ISupportsNavigation
         {
-            private readonly string _projectName;
+            protected override string GetProjectName()
+                => projectName;
 
-            public DefinitionItemEntry(
-                AbstractTableDataSourceFindUsagesContext context,
-                RoslynDefinitionBucket definitionBucket,
-                string projectName,
-                Guid projectGuid,
-                SourceText lineText,
-                MappedSpanResult mappedSpanResult)
-                : base(context, definitionBucket, projectGuid, lineText, mappedSpanResult)
+            protected override IList<Inline> CreateLineTextInlines()
+                => DefinitionBucket.DefinitionItem.DisplayParts.ToInlines(Presenter.ClassificationFormatMap, Presenter.TypeMap);
+
+            public bool CanNavigateTo()
             {
-                _projectName = projectName;
+                var workspace = document.Project.Solution.Workspace;
+                var documentNavigationService = workspace.Services.GetService<IDocumentNavigationService>();
+                return documentNavigationService != null;
```

Comment:
this should also do documentNavigationService.CanNavigateToSpan.

Note that IDocumentNavigationService will never be null.

Reply:
I don't think we should be calling CanNavigateToSpanAsync here. If this returned false it means VS will handle the navigation itself based on the file path and position. So if we determine that we can't navigate to the target and returned false, then we would say that VS can navigate to the target better then we can. But if we own the document/definition item then we have the full knowledge how to navigate to spans in it and VS won't have better knowledge.